packages/next/client/page-loader.js

```diff
@@ -76,6 +91,60 @@ export default class PageLoader {
     )
   }
 
+  /**
+   * @param {string} href the route href (file-system path)
+   * @param {string} asPath the URL as shown in browser (virtual path); used for dynamic routes
+   */
+  prefetchData(href, asPath) {
+    const { pathname: hrefPathname, query } = parse(href, true)
+    const { pathname: asPathname } = parse(asPath)
+
+    const route = normalizeRoute(hrefPathname)
+    return this.promisedSsgManifest.then(
+      (
+        s,
+        _dataHref,
+        _dynamic_regex,
+        _dynamic_groups,
+        _dynamic_matches,
+        _dynamic_built_route
```

Comment:
is defining these as args meant to avoid the extra bytes from declaring with `var`?

Reply:
Exactly

Reply:
Was just checking, seems like a neat trick to save some bytes

Reply:
Extracted for clarity at the cost of only few more bytes